src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -129,7 +129,6 @@ static void HandleNodeIdResolve(void * context, DnssdService * result, CHIP_ERRO
 static void HandleNodeBrowse(void * context, DnssdService * services, size_t servicesSize, CHIP_ERROR error)
 {
     ResolverDelegateProxy * proxy = static_cast<ResolverDelegateProxy *>(context);
-    proxy->Release();
 
     for (size_t i = 0; i < servicesSize; ++i)
     {
```

Comment:
Retain() can be called servicesSize number of times on the proxy object. Do we need to call Release() same number of times as well?

Reply:
This Release is not paired with those Retains.  We can and will enter this scope with a reference count of 1.  So for each of the Retains in the loop, there must be an additional Release in the call chain from within the loop.  The release happens in [HandleNodeResolve](https://github.com/project-chip/connectedhomeip/blob/c50811c1d355261664150550310a2596c970117b/src/lib/dnssd/Discovery_ImplPlatform.cpp#L45).

CC @vivien-apple